pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.hpp

```diff
@@ -636,10 +635,14 @@ template <class Precision> class StateVectorKokkos {
     ~StateVectorKokkos() {
         data_.reset();
         {
-            const std::lock_guard<std::mutex> lock(counts_mutex_);
-            counts_--;
-            if (counts_ == 0) {
-                Kokkos::finalize();
+            const std::lock_guard<std::mutex> lock(init_mutex_);
+            if (!is_exit_reg_) {
+                is_exit_reg_ = true;
+                std::atexit([]() {
```

Comment:
Do we need to add the <cstdlib> header as 'std::atexit()' is defined there?

Reply:
Good call --- done.